segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollector.java

```diff
@@ -343,6 +393,48 @@ void deleteBlockIndexEntriesForChunk(MetadataTransaction txn, String segmentName
         }
     }
 
+    /**
+     * Delete block index entries for given segment.
+     */
+    CompletableFuture<Void> deleteBlockIndexEntriesForSegment(String segmentName, long startOffset, long endOffset) {
+        val currentBatch = Collections.synchronizedSet(new HashSet<String>());
```

Comment:
Can you move this to line 408, in which case you don't need to clear it every time and don't need a synchronized set?

Reply:
done